src/components/Home/Home.js

```diff
@@ -41,12 +58,83 @@ function Home({ navigation }) {
     dispatch(fetchMovies(currentPage));
   };
 
-  useEffect(() => {
-    if (!movies.length) {
-      dispatch(fetchMovies());
+  const getDays = ms => {
+    if (!ms) {
+      return -1;
+    }
+    const minutes = Math.floor(ms / 60000);
+    const hours = Math.round(minutes / 60);
+    const days = Math.round(hours / 24);
+    return days;
+  };
+
+  const checkDataValidity = useCallback(() => {
+    let daysOld;
+    if (!lastFetchDate) {
+      daysOld = 0;
+      dispatch(setLastFecthDate(new Date()));
+    } else {
+      daysOld = getDays(new Date() - new Date(lastFetchDate));
+    }
+    if (daysOld > dataExpirationDays) {
+      dispatch(clearStore());
+      purge();
     }
+  }, [dataExpirationDays, dispatch, lastFetchDate]);
+
+  const fetchMoviesIfNeeded = useCallback(
+    (isConnected = true) => {
+      if (movies && !movies.length && isConnected) {
+        dispatch(fetchMovies());
+      }
+    },
+    [dispatch, movies]
+  );
+
+  const handleAppStateChange = useCallback(
+    nextAppState => {
+      if (
+        appState.current.match(/inactive|background/) &&
+        nextAppState === 'active'
+      ) {
```

Comment:
This is a weird syntax for an if, did the linter suggest this?

Reply:
linter 😂